internal/exec/runner.go

```diff
@@ -90,31 +90,58 @@ func newRunner(workDirPath string, input io.Reader, output io.Writer, options ..
 	return runner
 }
 
-func (r *runner) Version() error {
-	tabWriter := newTabWriter(r.output)
-	if _, err := fmt.Fprintf(tabWriter, "Version:\t%s\n", vars.Version); err != nil {
-		return err
-	}
-	if _, err := fmt.Fprintf(tabWriter, "Default protoc version:\t%s\n", vars.DefaultProtocVersion); err != nil {
-		return err
-	}
-	if _, err := fmt.Fprintf(tabWriter, "Go version:\t%s\n", runtime.Version()); err != nil {
-		return err
-	}
-	if vars.GitCommit != "" {
-		if _, err := fmt.Fprintf(tabWriter, "Git commit:\t%s\n", vars.GitCommit); err != nil {
+func (r *runner) Version(json bool) error {
+	out := &struct {
+		Version              string `json:"version"`
+		DefaultProtocVersion string `json:"default_protoc_version"`
+		GoVersion            string `json:"go_version"`
+		GitCommit            string `json:"git_commit"`
+		BuiltTimestamp       string `json:"built_timestamp"`
+		GOOS                 string `json:"GOOS"`
+		GOARCH               string `json:"GOARCH"`
+	}{
+		Version:              vars.Version,
+		DefaultProtocVersion: vars.DefaultProtocVersion,
+		GoVersion:            runtime.Version(),
+		GitCommit:            vars.GitCommit,
+		BuiltTimestamp:       vars.BuiltTimestamp,
+		GOOS:                 runtime.GOOS,
+		GOARCH:               runtime.GOARCH,
+	}
+
+	if json {
+		enc := json.NewEncoder(r.output)
+		enc.SetIndent("", "  ")
```

Comment:
Why not just use MarshalIndent and runner.println?

Reply:
@amckinney can’t remember what you did, but we should be consistent (and I think MarshalIndent would be the best across all json calls)

Reply:
I did this because it seemed like the simpler way to marshal to an `io.Writer` (`runner.output`).